examples/03_validation_set/plot_1_simple_crossval.py

```diff
@@ -0,0 +1,147 @@
+    "sentence reading vs checkerboard",
+    "sentence listening",
+    "calculation vs sentences",
+    "left vs right button press",
+    "checkerboard",
+]
+n_training_contrasts = 3
+
+brain_data = datasets.fetch_localizer_contrasts(
+    contrasts,
+    n_subjects=n_subjects,
+    get_anats=True,
+    get_masks=True,
+)
+
+source_imgs_paths = brain_data["cmaps"][0 : len(contrasts)]
+target_imgs_paths = brain_data["cmaps"][len(contrasts) : 2 * len(contrasts)]
+source_mask = brain_data["masks"][0]
+
+source_im = image.load_img(source_imgs_paths)
+target_im = image.load_img(target_imgs_paths)
+mask = image.load_img(source_mask)
+
+# %%
+# We then downsample the images by 5 to reduce the computational cost.
+SCALE_FACTOR = 5
+
+resized_source_affine = source_im.affine.copy() * SCALE_FACTOR
+resized_target_affine = target_im.affine.copy() * SCALE_FACTOR
+
+source_im_resized = image.resample_img(source_im, resized_source_affine)
+target_im_resized = image.resample_img(target_im, resized_target_affine)
+mask_resized = image.resample_img(mask, resized_source_affine)
+
+source_maps = np.nan_to_num(source_im_resized.get_fdata())
+target_maps = np.nan_to_num(target_im_resized.get_fdata())
+segmentation = mask_resized.get_fdata()
+
+coordinates = np.argwhere(segmentation > 0)
+
+source_features = source_maps[
+    coordinates[:, 0], coordinates[:, 1], coordinates[:, 2]
+].T
+target_features = target_maps[
+    coordinates[:, 0], coordinates[:, 1], coordinates[:, 2]
+].T
+
+fig = plt.figure()
+ax = fig.add_subplot(projection="3d")
+ax.scatter(coordinates[:, 0], coordinates[:, 1], coordinates[:, 2], marker="o")
+ax.view_init(10, 135)
+ax.set_axis_off()
+plt.show()
+
+# %%
+# We then compute the distance matrix between voxel coordinates.
+source_geometry = distance_matrix(coordinates, coordinates)
+target_geometry = source_geometry.copy()
+
+# %%
+# In order to avoid numerical errors when fitting the mapping, we normalize
+# both the features and the geometry.
+source_features_normalized = source_features / np.linalg.norm(
+    source_features, axis=1
+).reshape(-1, 1)
+target_features_normalized = target_features / np.linalg.norm(
+    target_features, axis=1
+).reshape(-1, 1)
+source_geometry_normalized = source_geometry / np.max(source_geometry)
+target_geometry_normalized = target_geometry / np.max(target_geometry)
+
+# %%
+# We now fit the mapping using the sinkhorn solver and 3 BCD iterations.
+mapping = FUGW(alpha=0.5, rho=1, eps=1e-4)
+_ = mapping.fit(
+    source_features=source_features_normalized[:n_training_contrasts],
+    target_features=target_features_normalized[:n_training_contrasts],
+    source_geometry=source_geometry_normalized,
+    target_geometry=target_geometry_normalized,
+    source_features_val=source_features_normalized[n_training_contrasts:],
+    target_features_val=target_features_normalized[n_training_contrasts:],
+    source_geometry_val=source_geometry_normalized,
+    target_geometry_val=target_geometry_normalized,
```

Comment:
Maybe you should say a word about the fact that `{source,target}_geometry_val` are given the same values as `{source,target}_geometry`.
`{source,target}_geometry_val` could be omitted here, and maybe you could say that it's hard to find a brain-related example where these should be different at validation time than at training time.